typeclasses/npcshop/npcshop.py

```diff
@@ -257,13 +258,16 @@ def at_object_creation(self):
         self.cmdset.add_default(StoreRoomCmdSet)
 
     def return_appearance(self, caller):
-        wares = []
-        for item in self.contents:
-            if item.db.value:
-                wares.append(item.key)
+        wares = [item for item in self.contents if item.db.value]
         desc = super(StoreRoom, self).return_appearance(caller)
         for line in desc.split("\n"):
-            if line.strip().split("(")[0] in wares:
-                line = "|y" + line
+            if any(line.find(ware.key) > -1 for ware in wares):
+                try:
+                    line = "|y{line} |n({price})".format(
```

Comment:
I think this is a worthy intention, but the problem is that the parent's `return_appearance` is monolithic. There's no way to 'call' the components you list outside of just calling `output = super(...).return_appearance()` and parsing `output`, which obviously has its pitfalls.

One question about this: how important is it really for `StoreRoom`s to track with other changes to `Room`? It is, in a sense, a highly-specific utility room, rather than a backdrop for game action. It may not be a problem if the output eventually looks a little "legacy" and/or needs a manual update to stay in sync with the parent. 

Another consideration is the source of these potential upstream changes. I would guess that most changes to `return_appearance` in Ainneve will be initiated in this code base, rather than because upstream Evennia has changed how the vanilla `DefaultRoom` class does things. This means we'll have the opportunity to evaluate and port changes if they are important.

The more I think about it, the more I think we just bite the bullet and copy/modify the base `return_appearance`. However, if you can come up with something more elegant, I'm still totally wiling to entertain the idea.